apps/session-manager/src/controllers/__tests__/zendesk.spec.ts

```diff
@@ -0,0 +1,244 @@
+    )();
+
+    expect(E.isRight(response)).toBeTruthy();
+    expect(res.json).toHaveBeenCalledWith({
+      jwt: aZendeskSupportToken,
+    });
+  });
+
+  it("should return an IResponseErrorInternal if user does not have any email address", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(
+        ResponseSuccessJson({
+          ...mockedInitializedProfile,
+          email: undefined,
+          is_email_validated: false,
+          spid_email: undefined,
+        }),
+      ),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Profile has not a validated email address",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return an IResponseErrorInternal if Profile has not a validated email address", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(
+        ResponseSuccessJson({
+          ...mockedInitializedProfile,
+          is_email_validated: false,
+        }),
+      ),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Profile has not a validated email address",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return an IResponseErrorInternal if user has only the spid email", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(
+        ResponseSuccessJson({
+          ...mockedInitializedProfile,
+          email: undefined,
+          is_email_validated: false,
+        }),
+      ),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Profile has not a validated email address",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getJwtZendeskSupportToken returns an Error", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(ResponseSuccessJson(mockedInitializedProfile)),
+    );
+
+    mockGetZendeskSupportToken.mockReturnValue(
+      TE.left(new Error("ERROR while generating JWT support token")),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error("ERROR while generating JWT support token"),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getProfile returns an error", async () => {
+    mockGetProfile.mockReturnValue(TE.left(Error("")));
```

Comment:
I'd prefer to specify an error message.

Reply:
done in https://github.com/pagopa/io-auth-n-identity-domain/pull/34/commits/15b5283dc8882ae73e35c2dc05e3f91746718016